server/src/main/java/com/cloud/servlet/ConsoleProxyServlet.java

```diff
@@ -417,6 +425,40 @@ private String composeThumbnailUrl(String rootUrl, VirtualMachine vm, HostVO hos
         return sb.toString();
     }
 
+    /**
+     * Sets the URL to establish a VNC over websocket connection
+     */
+    private void setWebsocketUrl(VirtualMachine vm, ConsoleProxyClientParam param) {
+        String ticket = acquireVncTicketForVmwareVm(vm);
```

Comment:
_ acquireVncTicketForVmwareVm()_ returns null, in case of any errors/expections, check for the same ?